test/utils.compareBlockNumbers.js

```diff
@@ -37,13 +41,32 @@ const tests = [
     { input: {a: genesis, b: new BN(2)}, result: -1 },
     { input: {a: genesis, b: latest}, result: -1 },
     { input: {a: genesis, b: pending}, result: -1 },
+    { input: {a: earliest, b: finalized}, result: -1 },
+    { input: {a: genesis, b: finalized}, result: -1 },
+    { input: {a: earliest, b: safe}, result: -1 },
+    { input: {a: genesis, b: safe}, result: -1 },
     // latest
     { input: {a: latest, b: 0}, result: 1 },
     { input: {a: latest, b: new BN(1)}, result: 1 },
     { input: {a: latest, b: pending}, result: -1 },
     // pending 
     { input: {a: pending, b: 0}, result: 1 },
     { input: {a: pending, b: new BN(1)}, result: 1 },
+    { input: {a: finalized, b: 0}, result: 1 },
+    { input: {a: finalized, b: new BN(1)}, result: 1 },
+    // finalized
+    { input: {a: finalized, b: 0}, result: 1 },
+    { input: {a: finalized, b: new BN(1)}, result: 1 },
+    { input: {a: finalized, b: pending}, result: -1 },
+    { input: {b: finalized, a: 0}, result: -1 },
+    { input: {b: finalized, a: new BN(1)}, result: -1 },
+    { input: {b: finalized, a: pending}, result: 1 },
+    // safe
+    { input: {a: safe, b: 0}, result: 1 },
+    { input: {a: safe, b: pending}, result: -1 },
+    { input: {b: safe, a: 0}, result: -1 },
+    { input: {b: safe, a: pending}, result: 1 },
+
```

Comment:
I think this test case would fail, won't it? Or what is the expected value for such a comparison?
```suggestion
    { input: {b: safe, a: 2}, result: -1 },
```

Reply:
This test was added at start of PR after that I think I added that this function doesn't support num vs safe tag in comments.

Reply:
it should fail yes, as function will return undefined now in this case.